README.md

```diff
@@ -152,6 +168,10 @@ the URL back to the app.
    If the user has multiple browsers installed, he still has to
    choose between them.
 
+   Note that (in sharp contrast to iOS) the Google Play Store allows any app to claim to be a browser. 
+
+   JJH: Am I correct in this note? The wording of the previous paragraph is making me doubt myself.
```

Comment:
I'm not exactly sure what you have to do to publish a browser in the Google Play Store, but it can't be that hard because there are a lot of different browsers. [Here](https://developer.android.com/guide/topics/permissions/default-handlers#follow-requirements-default-handlers) I found a general list of how an app can become a default handler for a task. How does the process of publishing a browser on iOS work?

Reply:
Thanks for the link! I'm pretty unclear if any manual review by play store staff is required.

It's clear that in some cases there's very strict rules that involve manual review:

https://android-developers.googleblog.com/2019/01/reminder-smscall-log-policy-changes.html

but it's not obvious if this applies to browser or not. I think I might update my suggested text to just say that it's unclear what checking play does?

For iOS the process is documented here:

https://developer.apple.com/documentation/xcode/allowing_apps_and_websites_to_link_to_your_content/preparing_your_app_to_be_the_default_browser_or_email_client

the developer needs to be explicitly granted an extra signing permission, and it seems this has to be granted manually by Apple in response to an email exchange - and as normal, each published version of the app also has to pass a manual assessment by the Apple App Store Review Team, whereas the google play checks are generally 100% automated and I think it's still generally viewed as fairly straightforward to get a malicious app onto play, so long as you don't try and touch any of the highly restricted permissions like access to read all the user's sms messages.

Reply:
I'm also not sure if there is a manual review process for browsers in the Play Store. I have never published an app in the Play Store but [this article](https://www.androidpolice.com/2019/09/03/google-play-expands-store-app-review/) suggests that a lot of apps are manually reviewed. I think it is a good idea to say that the actual checks are unclear!

But the proposed code in the section **RP App to IDP** uses the browser selection method from the AppAuth-Android project so we always launch a verified browser.

Reply:
Yes, it's unclear. I think possibly new apps might be manually reviewed. We seem to see most/all our google play app updates get published without obvious manual review. Another article I saw hints that some kind of automated process might flag some app updates if you change particular things for manual review.

My latest text is:

> Note that in the Google Play Store is it not clear how hard it is to publish an app that provides CATEGORY_APP_BROWSER (whereas on iOS a signing permission manually granted by Apple is required for an app to be selectable as the default browser).

Reply:
I think this latest text sums it up very well.